### DISCUSSION TITLE
Rendering a scatterplot in html output

### DISCUSSION THREAD

**Discussion by KennethGottfredsen** [quarto-dev/quarto-cli#2622] [Q&A] [answered]
Hi guys, 

I want to show this scatterplot in the viewer output, when I press the rend button.  At the moment it only shows the plot under the chunk. Is there a chunk options that allows me to do that? In order to show it in the html or pdf output im saving the image and pasting it under dude chunk text, which is not optimal when the document is huge. 

Below is the code: 

{r}
#| eval: FALSE
#| cache: TRUE
ggplot(data = mpg) + 
  geom_point(mapping = aes(x = displ, y = cty), color = "green")

/ Kenneth.

**Comment by rgaiacs** [comment]
Hi @KennethGottfredsen, next to the "Render" button, you have the "Gear" button. Under the "Gear" button you will find the option that you want.

![output](https://user-images.githubusercontent.com/1506457/192467663-b4bd153b-28ec-4406-81a2-a99be6e83a90.png)

Check "Chunk Output in Console".

**Reply by KennethGottfredsen** [reply]
I want it to display the scatterplot in the html and pdf output together with the text in the viewer pane. With that option I'm only able to see the entire picture in the console. So when I render it, the image is still missing in the viewer.

 At the moment I'm saving all the scatterplots as images on my computer, and inserting them manually below the code chunk. Thank you for the help tho. Do you know what I mean?

**Comment by cderv** [comment]
@KennethGottfredsen can you share a qmd source code that reproduce the issue ? 

From your subset of code above, you have
````
#| eval: FALSE
````

This means that when you render, the chunk won't be evaluated, so you'll only have the source code chunk in the output, as you described. 

You probably don't want this option to FALSE if you want the plot to show in your output

**Reply by KennethGottfredsen** [reply]
Hi,

The thing is, I want the html output to show my code. That's why I have eval: FALSE. If I set the option to TRUE, I won't be able to render the html output at all. If I have no code options enabled, I get an error saying:

 "Quitting from lines 233-253 (kapitel_3_R4D_opgaver.qmd) 
Fejl i ggplot(data = mpg) : could not find function "ggplot"
Kald: .main ... withVisible -> eval_with_user_handlers -> eval -> eval"


When you say cmd source code I guess you mean the following:
`
#| eval: FALSE

# Geom er et geometrisk objekt som et plot repræsenterer data med. 
# Alle funktionerne i geom har mapping som et argument, det er dog ikke alle aesthethics der virker. 

ggplot(data = mpg) + 
  geom_point(mapping = aes(x = displ, y = hwy))
`

/ Kenneth

**Reply by KennethGottfredsen** [reply]
This is an example of me writing TRUE instead of FALSE. 

#| eval: TRUE
ggplot(data = mpg) + 
  geom_point(mapping = aes(x = displ, y = hwy))

**Comment by cderv** [comment, marked as answer]
> The thing is, I want the html output to show my code. That's why I have eval: FALSE.

For this you need `echo: true` to show code output. This is the default, `eval: false` means don't evaluate the code. 

Do you want to evaluate de R code ? Do you want the plot in the HTML output ? 
If so you need to activate evaluation for the chunk. 

> If I set the option to TRUE, I won't be able to render the html output at all. If I have no code options enabled, I get an error saying:
>
> ```
> "Quitting from lines 233-253 (kapitel_3_R4D_opgaver.qmd)
> Fejl i ggplot(data = mpg) : could not find function "ggplot"
> Kald: .main ... withVisible -> eval_with_user_handlers -> eval -> eval"
> ```

You see this in the error message you have
```
could not find function "ggplot"
```

This means  the R function is not found, so probably that `library(ggplot2)` has not be called in a previous chunk. 

Please to read about Get Started for R with Quarto to understand how this works. https://quarto.org/docs/get-started/hello/rstudio.html


And next time, please provide a full reproducible example. Thank you

**Reply by KennethGottfredsen** [reply]
Wow thank you! This sorted out all my problems, mostly lol. But you were right, I forgot to load the ggplot2 with the library function. 

I did get a few warnings in the other code chunks, but I did warning: FALSE and that sorted out the problem. 

When you say reproducible example, do you mean I should upload my qmd file here?

**Reply by cderv** [reply]
Glad it works now. 

> When you say reproducible example, do you mean I should upload my qmd file here?

Yes or just copy paste the qmd source using correct formatting so that we can copy paste and run on our side. Think about it next time! :)

**Reply by KennethGottfredsen** [reply]
Sure! Thank you.
